tests/framework/statistics/metadata.py

```diff
@@ -0,0 +1,155 @@
+        """Return measurement dictionary."""
+
+
+# pylint: disable=R0903
+class DictProvider(Provider):
+    """Backend for test metadata retrieval."""
+
+    UNIT_KEY = "unit"
+    STATISTICS_KEY = "statistics"
+
+    def __init__(self,
+                 measurements: dict,
+                 baseline_provider: BaselineProvider):
+        """
+        Initialize metadata provider.
+
+        The provider expects to receive measurements following the below
+        schema:
+        ```
+        "measurements": {
+            "$id": "MEASUREMENTS_SCHEMA"
+            "type": "object",
+            "definitions": "definitions": {
+                "Criteria": {
+                    "type": "string",
+                    "description": "Comparison criteria class name. They are
+                    implemented in the `statistics.criteria` module."
+                }
+                "Function": {
+                    "type": "string",
+                    "description": "Statistic functions class name. They are
+                    implemented in the `statistics.function` module."
+                }
+                "StatisticDef": {
+                    {
+                        "type": "object",
+                        "description": "Exhaustive statistic definition."
+                        "properties": {
+                            "name":     { "type": "string" },
+                            "function": {
+                                "type": "string"
+                                "$ref": "#/definitions/Function"
+                            },
+                            "criteria": {
+                                "type": "string"
+                                "$ref" "#/definitions/Criteria"
+                            }
+                        },
+                        "required": ["function"]
+                    }
+                }
+            },
+            "properties": {
+                "key": {
+                    "type": "string",
+                    "description": "Measurement name."
+                },
+                "value": {
+                    "type": "object",
+                    "properties": {
+                        "unit": "string",
+                        "statistics": {
+                            "type": "object",
+                            "properties": {
+                                "key": {
+                                    "type": "string",
+                                    "description": "Statistic name."
+                                },
+                                "value": {
+                                    "type": "object",
+                                    "$ref": "#/definitions/StatisticDef"
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        ```
+        """
+        super().__init__()
+        for ms_name in measurements:
```

Comment:
This would mean naming the logic that initialize the measurements dict, from the constructor, in a way which will indicate about what it does. I personally do not feel that a dedicated abstract method should be added for this case at this specific moment and I would wait to see if such interface is really needed, for other implementations as well.

Reply:
I'm fine keeping it this way for now